.github/workflows/storybook_deploy_main.yml

```diff
@@ -0,0 +1,27 @@
+name: Deploy Main Storybook
+
+on:
+  push:
+    branches:
+      - develop
+    paths:
+      - '**[Ss]tories**'
+
+jobs:
+  deploy_main_storybook:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run npm install
+      run: npm install
+    - name: Build Storybook
+      run: npm run build-storybook -- --output-dir=storybooks/main
+    - uses: yg/deploy-surge@v1.0.1
+      name: Deploy Storybook to surge.sh
+      with:
+        SURGE_TOKEN: ${{ secrets.SURGE_TOKEN }}
+        DOMAIN_NAME: foreman.surge.sh
```

Comment:
I'd also keep this as a secret tbh. This can allow people to easily set their own domain in a fork to show their own storybook if they want to.